core/src/nodes/listeners.rs

```diff
@@ -220,9 +221,48 @@ where
 #[cfg(test)]
 mod tests {
     extern crate libp2p_tcp_transport;
+
     use super::*;
     use transport;
     use tokio::runtime::current_thread::Runtime;
+    use std::io;
+    use futures::{future::{self}, stream};
+    use tests::dummy_transport::{DummyTransport, ListenerState};
+
+    // Test helper that lets us poke in innards of individual `Listener`s and
+    // set things up for the tests.
+    impl ListenersStream<DummyTransport> {
+        fn set_listener_state(&mut self, idx: usize, state: ListenerState) {
+            let mut l = self.listeners.remove(idx);
+            l.listener =
+                match state {
+                    ListenerState::Error => {
+                        let stream = stream::poll_fn(|| future::err(io::Error::new(io::ErrorKind::Other, "oh noes")).poll() );
+                        Box::new(stream)
+                    }
+                    ListenerState::Ok(async) => {
+                        match async {
+                            Async::NotReady => {
+                                let stream = stream::poll_fn(|| future::empty().poll() );
+                                Box::new(stream)
+                            }
+                            Async::Ready(Some(_)) => {
+                                let addr = l.address.clone();
+                                let stream = stream::poll_fn(|| future::ok(Some(1)).poll() )
+                                    .map(move |stream| future::ok( (stream, future::ok(addr.clone())) ));
+                                Box::new(stream)
+
+                            }
+                            Async::Ready(None) => {
+                                let stream = stream::poll_fn(|| future::ok(None).poll() );
+                                Box::new(stream)
+                            }
+                        }
+                    }
```

Comment:
Should the `ListenerState::Ok` branch of this `match` not be doing the same as in `DummyTransport::listen_on`?

Reply:
Definitely should, fixing.